File: packages/ts/src/accounts/pqc-algorithm.ts
/**
 * PQC algorithm identifiers, mirroring the chain's `x/pqc` cryptographic-agility
 * framework (core `x/pqc/types/algorithm.go`).
 *
 * These numeric IDs are the on-the-wire values the chain expects in
 * `MsgRegisterPQCKeyV2.algorithm_id` and in the `PQCHybridSignature` TX
 * extension's `algorithm_id` field, so they MUST stay in sync with the core enum.
 */

/**
 * Identifies a PQC algorithm. Numeric to match the chain's `uint32 AlgorithmID`.
 * IDs 1–2 are the initial algorithms; 3–255 are reserved for future
 * governance-approved algorithms.
 */
export type AlgorithmID = number;

/** Unset / invalid algorithm (core: `AlgorithmUnspecified`). */
export const AlgorithmUnspecified: AlgorithmID = 0;
/** Dilithium-5 = ML-DSA-87, NIST FIPS 204 signatures (core: `AlgorithmDilithium5`). */
export const AlgorithmDilithium5: AlgorithmID = 1;
/** ML-KEM-1024, NIST FIPS 203 key encapsulation (core: `AlgorithmMLKEM1024`). */
export const AlgorithmMLKEM1024: AlgorithmID = 2;

/** Human-readable name for an algorithm ID (matches core `AlgorithmID.String()`). */
export function algorithmName(id: AlgorithmID): string {
  switch (id) {
    case AlgorithmUnspecified:
      return "unspecified";
    case AlgorithmDilithium5:
      return "dilithium5";
    case AlgorithmMLKEM1024:
      return "mlkem1024";
    default:
      return `algorithm_${id}`;
  }
}

/** True if the algorithm is a digital-signature scheme (core `IsSignature`). */
export function isSignatureAlgorithm(id: AlgorithmID): boolean {
  return id === AlgorithmDilithium5;
}

File: packages/java/src/main/java/io/github/qorechain/pqc/PqcAlgorithm.java
package io.github.qorechain.pqc;

/**
 * PQC algorithm identifiers, mirroring the chain's cryptographic-agility
 * framework. These numeric IDs are the on-the-wire values the chain expects in
 * {@code MsgRegisterPQCKeyV2.algorithm_id} and the {@code PQCHybridSignature}
 * TX extension, so they MUST stay in sync with the core enum.
 */
public final class PqcAlgorithm {

    private PqcAlgorithm() {}

    /** Unset / invalid algorithm. */
    public static final int ALGORITHM_UNSPECIFIED = 0;
    /** Dilithium-5 = ML-DSA-87, NIST FIPS 204 signatures. */
    public static final int ALGORITHM_DILITHIUM5 = 1;
    /** ML-KEM-1024, NIST FIPS 203 key encapsulation. */
    public static final int ALGORITHM_MLKEM1024 = 2;

    /** Human-readable name for an algorithm ID. */
    public static String algorithmName(int id) {
        switch (id) {
            case ALGORITHM_UNSPECIFIED:
                return "unspecified";
            case ALGORITHM_DILITHIUM5:
                return "dilithium5";
            case ALGORITHM_MLKEM1024:
                return "mlkem1024";
            default:
                return "algorithm_" + id;
        }
    }

    /** True if the algorithm is a digital-signature scheme. */
    public static boolean isSignatureAlgorithm(int id) {
        return id == ALGORITHM_DILITHIUM5;
    }
}
